esp/logging/logginglib/loggingagentbase.hpp

```diff
@@ -249,14 +250,67 @@ class CUpdateLogRequestWrap : implements IEspUpdateLogRequestWrap, public CInter
     void setNoResend(bool val) { noResend = val; };
 };
 
+interface IEspLogAgentVariant : extends IInterface
+{
+    virtual const char* getName() const = 0;
+    virtual const char* getType() const = 0;
+    virtual const char* getGroup() const = 0;
+};
+
+struct IEspLogAgentVariantComparator
+{
+    // Implement alternative to std::less for potential set ordering
+    bool operator () (const Owned<const IEspLogAgentVariant>& lhs, const Owned<const IEspLogAgentVariant>& rhs) const
+    {
+        return (*this)(lhs.get(), rhs.get());
+    }
+    bool operator () (const IEspLogAgentVariant* lhs, const IEspLogAgentVariant* rhs) const
+    {
+        return compare(lhs, rhs) < 0;
```

Comment:
Why < 0?

Reply:
Sets determine ordering using an equivalent of '<' to decide if lhs should precede rhs. The compare method, like stricmp that it calls, returns a negative number when lhs precedes rhs and the operator result is true when a negative number is returned.